### DISCUSSION TITLE
Are peer dependency install incorrectly?

### DISCUSSION THREAD

**Discussion by unional** [pnpm/pnpm#8800] [Q&A]
Say a package `pkg-a` has this `package.json`:

```jsonc
{
  "name": "pkg-a",
  "version": "1.0.0",
  "peerDependencies": {
    "pkg-b": "^1.0.0"
  }
}
```

When installing `pkg-a`, `pnpm` yields:

```sh
node_modules
  .pnpm
    pkg-a@1.0.0...
      pkg-b # link to `.pnpm/pkg-b@1.0.0...`
    pkg-b@1.0.0...
  pkg-a # link to `.pnpm/pkg-a@1.0.0...`
```

i.e. `import 'pkg-b'` will fail.
On one hand, this is correct because the project didn't install `pkg-b` directly.

But on the other hand, this is wrong because `pkg-b` supposed to be a "peer" of `pkg-a`. That's what `pkg-a` expects thus declaring `pkg-b` as a peer dependency instead of a regular dependency.

This become an issue when a plugin tells the plugin host to load another plugin.

Examples of plugin host: `vite`, `storybook`.

Right now, the consuming project needs to install the `peerDependencies` in `pkg-a` manually in order for the plugin host (`vite` or `storybook`) to load the plugins correctly.
i.e., the `auto-peers-install` is not "working".

Cross linking the discussion on `storybook`:
https://github.com/storybookjs/storybook/discussions/29692

**Comment by zkochan** [comment]
`pkg-b` is not a dependency of your project, so it isn't in the root of node_modules. If you need access to `pkg-b` you should install it as a direct dependency of the project where you want to import it. `auto-install-peers` works because as you have shown in your example, it is linked to `pkg-a`. If you will install it as a regular dependency, it won't be duplicated, the peer will be resolved from the version that you installed.

**Comment by unional** [comment]
So does to mean that if the consuming project wants to install the dependencies as `pkg-a` intended, it should:

- set `auto-install-peers` to `false`
- set `strict-peer-dependencies` to `true`
- install all peer dependencies manually

The current way of installing those peer dependencies is not how peer dependencies meant to be installed, IMO.

At the same time, I do understand that this is more like a workaround/helping features that will cover some (common?) cases, while avoid monkeying user's `package.json`.

[The documentation](https://pnpm.io/npmrc#auto-install-peers) didn't get into these details giving the impression that "it will just work".
